replica/table.cc

```diff
@@ -1786,7 +1786,10 @@ future<> table::perform_cleanup_compaction(compaction::owned_ranges_ptr sorted_o
         co_await flush();
     }
 
-    auto& cg = *get_compaction_group(0);
+    if (compaction_groups().size() != 1 || !storage_groups()[0]) {
+        on_internal_error(tlogger, "table::perform_cleanup_compaction: single compaction group expected");
+    }
```

Comment:
I think we can safely assume compaction_groups() will be of size 1 with static sharding.

Also, I plan on removing unique_ptr usage in storage group map, so latter condition is not needed.